numpyro/infer/sa.py

```diff
@@ -172,7 +172,8 @@ def sample_kernel(sa_state, model_args=(), model_kwargs=None):
             log_weights_ = dist.MultivariateNormal(locs_, scale_tril=scales_).log_prob(zs_) + pes_
         else:
             log_weights_ = dist.Normal(locs_, scales_).log_prob(zs_).sum(-1) + pes_
-        log_weights_ = jnp.where(jnp.isnan(log_weights_), -jnp.inf, log_weights_)
+        # mask invalid values (nan, +inf) by -inf
+        log_weights_ = jnp.where(jnp.isfinite(log_weights_), log_weights_, -jnp.inf)
```

Comment:
LGTM, but when would this ever be `+inf`?

Reply:
It comes from a diffusion drift model in our forum (where the model has lax.scan). I guess something like 1/0 is involved or precision loss during the loop... (the model is pretty complicated and involves sequential matrix multiplication....).